src/cultionet/models/base_layers.py

```diff
@@ -408,71 +742,341 @@ def __init__(
     def forward(self, x: torch.Tensor) -> torch.Tensor:
         q = self.query(x)
         k = self.key(x)
-        v = self.values(x)
+        v = self.value(x)
 
         attention_spatial = self.spatial_sim(q, k)
         v_spatial = attention_spatial * v
 
         attention_channel = self.channel_sim(q, k)
         v_channel = attention_channel * v
 
-        v_channel_spatial = (v_spatial + v_channel) * 0.5
-        v_channel_spatial = self.norm(v_channel_spatial)
+        attention = (v_spatial + v_channel) * 0.5

```

Comment:
is this related to the attention gate you mentioned at one point in your model architecture?

Reply:
It's similar, but a different attention method than what you're referring to.